### PR TITLE
libkv support

### PR DESCRIPTION
**Pull request by EmileVauge**
This PR adds support to libkv in place of native kv client libraries.
It also adds support to etcd, zookeeper and boltdb.

https://github.com/EmileVauge/traefik/issues/25
https://github.com/EmileVauge/traefik/issues/9

### CODE REVIEW

**Comment by EmileVauge** [comment]
There will be another commits to update the doc :)

**Comment by EmileVauge** [comment]
@vdemeester, the ball is in your court ;)

**Comment by vdemeester** [comment]
I know :sweat_smile: :wink:

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -39,6 +43,19 @@
...
+          "Rev": "cc90c7b091275e606ad0ca7102a23fb2072f3f5e"
+        },
+        {
+          "ImportPath": "github.com/davecgh/go-spew/spew",
+          "Rev": "2df174808ee097f90d259e432cc04442cf60be21"
+        },
+        {
+          "ImportPath": "github.com/docker/libkv",
+          "Rev": "3732f7ff1b56057c3158f10bceb1e79133025373"
+        },
```
Comment (vdemeester) [inline Godeps/Godeps.json]: Indentation is not good :sweat: (I'm OCD so don't worry about it :cat:)

File: Makefile [hunk 2]
```diff
@@ -47,6 +47,9 @@ validate-govet: build
 build: dist
 	docker build -t "$(TRAEFIK_DEV_IMAGE)" -f build.Dockerfile .
 
+build-no-cache: dist
+	docker build --no-cache -t "$(TRAEFIK_DEV_IMAGE)" -f build.Dockerfile .
```
Comment (vdemeester) [inline Makefile]: Good idea :kissing_cat:

**Comment by vdemeester** [comment]
Overall **looks good** and **green**, so I like it. Question though : I really feel we should make more packages (e.g. in the context of this PR `provider`), WDYT ? (can be done in another PR not problem, so it's no big deal)

**Comment by EmileVauge** [comment]
Yep, will reorganise the packages a bit in another PR :) Corrections made!

**Comment by vdemeester** [comment]
@EmileVauge ok cool :+1: 
I'll take a look on the reorg' too, so ping me when you do that (so we don't do the same at the same time :stuck_out_tongue_closed_eyes:).

Here is your LGTM and merge 👼 !

**Pull request merged**